server/src/domain/repositories/asset-stack.repository.ts

```diff
@@ -0,0 +1,10 @@
+import { AssetStackEntity } from '@app/infra/entities/asset-stack.entity';
+
+export const IAssetStackRepository = 'IAssetStackRepository';
+
+export interface IAssetStackRepository {
+  create(assetStack: Partial<AssetStackEntity>): Promise<AssetStackEntity>;
+  save(asset: Pick<AssetStackEntity, 'id'> & Partial<AssetStackEntity>): Promise<AssetStackEntity>;
```

Comment:
```suggestion
  update(asset: Pick<AssetStackEntity, 'id'> & Partial<AssetStackEntity>): Promise<AssetStackEntity>;
```
In general, the repos try to have get, create, update, and delete. Although it's fine to have create and update both call "save" internally.